### PR TITLE
Add HAProxy Ingress as a downstream implementation

### PR DESCRIPTION
**Pull request by jcmoraisjr**
---
labels: 'tide/merge-method-squash'
---

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Add HAProxy Ingress to the downstream implementations list

**Which issue(s) this PR fixes**:

none

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: site-src/references/implementations.md [hunk 1]
```diff
@@ -10,17 +10,19 @@ Implementors of Gateway API are encouraged to update this document with status i
 - [Emissary-Ingress (Ambassador API Gateway)][2] (alpha)
 - [Gloo Edge 2.0][3] (work in progress)
 - [Google Kubernetes Engine][4] (public preview)
-- [Istio][5] (alpha)
-- [Kong][6] (work in progress)
-- [Traefik][7] (alpha)
+- [HAProxy Ingress](5) (alpha)
```
Comment (robscott) [inline site-src/references/implementations.md:13]:
```suggestion
- [HAProxy Ingress][5] (alpha)
```

File: site-src/references/implementations.md [hunk 2]
```diff
@@ -70,6 +72,15 @@ Balancing][gclb] is available in [public preview][gke-gateway].
 [gclb]:https://cloud.google.com/load-balancing
 [gke-gateway]: https://cloud.google.com/kubernetes-engine/docs/concepts/gateway-api
 
+### HAProxy Ingress
+
+[HAProxy Ingress](h1) is a community driven ingress controller implementation for HAProxy.
```
Comment (robscott) [inline site-src/references/implementations.md:77]:
```suggestion
[HAProxy Ingress][h1] is a community driven ingress controller implementation for HAProxy.
```

File: site-src/references/implementations.md [hunk 3]
```diff
@@ -70,6 +72,15 @@ Balancing][gclb] is available in [public preview][gke-gateway].
...
+
+HAProxy Ingress v0.13 partially supports the Gateway API's v1alpha1 specification. See the [controller's Gateway API documentation](h2) to get informed about conformance and roadmap.
```
Comment (robscott) [inline site-src/references/implementations.md:79]:
```suggestion
HAProxy Ingress v0.13 partially supports the Gateway API's v1alpha1 specification. See the [controller's Gateway API documentation][h2] to get informed about conformance and roadmap.
```

**Review by robscott — COMMENTED** [review]
Awesome to see support from HAProxy Ingress, thanks for the PR! Looks like this just a few tiny changes to markdown links but otherwise LGTM. For reference, deploy preview for this PR is available at https://deploy-preview-692--kubernetes-sigs-gateway-api.netlify.app/references/implementations/.

File: site-src/references/implementations.md (hunk 1 above)
Comment (jcmoraisjr) [reply site-src/references/implementations.md]: Oh sure, my bad, used to use the `[text](link)` syntax and didn't see the difference. Fixed now.

Comment (robscott) [reply site-src/references/implementations.md]: No worries, always throws me off too, thanks for the quick update!

**Comment by robscott** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**